src/routes/auth/guards/auth.guard.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Inject,
+  Injectable,
+} from '@nestjs/common';
+import { IAuthRepository } from '@/domain/auth/auth.repository.interface';
+import { AuthService } from '@/routes/auth/auth.service';
+
+/**
+ * The AuthGuard should be used to protect routes that require authentication.
+ *
+ * It checks for the presence of a valid JWT access token in the request and
+ * verifies its validity before allowing access to the route.
+ *
+ * 1. Check for the presence of an access token in the request.
+ * 2. Verify the token's validity.
+ * 3. If valid, allow access.
+ */
+
+@Injectable()
+export class AuthGuard implements CanActivate {
+  constructor(
+    @Inject(IAuthRepository) private readonly authRepository: IAuthRepository,
+  ) {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest();
+
+    const accessToken = this.authRepository.getAccessToken(
+      request,
+      AuthService.AUTH_TOKEN_TOKEN_TYPE,
```

Comment:
nitpick: out of context for this PR – we should probably move this to the `AuthRepository` as it's used by both the domain and routes.

Reply:
Agreed. As we will migrate to using cookies, this will ultimately be removed so I will merge this as is for now.